### PR TITLE
feat: add support for multiple eth rpc endpoints

### PR DESCRIPTION
**Pull request by jim380**
Resolves Issue https://github.com/umee-network/peggo/issues/196

### CODE REVIEW

File: cmd/peggo/flags.go [hunk 1]
```diff
@@ -99,11 +109,116 @@ func ethereumOptsFlagSet() *pflag.FlagSet {
...
+	return func() *ethclient.Client {
+		for {
+			rawEndpoints := rawEndpoints
+			endpoints, working, err := ethereumRPCValidate(rawEndpoints)
+			if err != nil {
+				log.Println(err)
+			}
+			log.Printf("%d of the provided Ethereum rpc endpoints are currently working", working)
+			if connectionErrors > len(endpoints) {
+				log.Println("too many connection errors, please double check your endpoints and try again")
```
Comment (toteki) [inline cmd/peggo/flags.go:166]: Should a nil return or similar be added to break out of the for loop after logging this message?

File: cmd/peggo/flags.go [hunk 2]
```diff
@@ -99,11 +109,116 @@ func ethereumOptsFlagSet() *pflag.FlagSet {
...
+			}
+			ethRPC, err := ethClientFetch(endpoints)
+			if err != nil {
+				connectionErrors += 1
+				log.Println("failed to dial rpc node: %w", err)
+				continue
```
[the remark this answers is not in the document]
Comment (jim380) [reply cmd/peggo/flags.go]: good catch. fixed in the latest commit.

File: cmd/peggo/flags.go (hunk 1 above)
Comment (jim380) [reply cmd/peggo/flags.go]: I added one in the latest commit.

File: cmd/peggo/flags.go [hunk 3]
```diff
@@ -99,11 +109,116 @@ func ethereumOptsFlagSet() *pflag.FlagSet {
...
+		log.Println("could not start eth client:", err)
+		return nil, err
+	}
+	log.Println("connected to", endpoint)
+	return client, nil
+}
+
+func ethClientConnect(rawEndpoints string) *ethclient.Client {
+	connectionErrors := 0
+	return func() *ethclient.Client {
```
[the remark this answers is not in the document]
Comment (jim380) [reply cmd/peggo/flags.go]: Correct. I originally wanted to put the anonymous functions in goroutine. Then I forgot to after shifting some code around. In the latest commit I removed the anonymous function within the connect function itself, and instead wrapped the connect  function in goroutine when called.

**Comment by alexanderbez** [comment]
Thanks for the contribution @jim380! But after further evaluation, I think this problem is actually a bit more complex than we thought and we're currently considering an approach at the nginx proxy level.

**Pull request closed without merging**